packages/atomic/src/components/search/tabs/atomic-tab-manager/atomic-tab-manager.tsx

```diff
@@ -0,0 +1,97 @@
+import {TabManager, TabManagerState, buildTabManager} from '@coveo/headless';
+import {Component, h, Element, State, Prop} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {getActiveTab} from '../../../../utils/tab-utils';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-tab-manager',
+  styleUrl: 'atomic-tab-manager.pcss',
+  shadow: true,
+})
+export class AtomicTabManager {
+  @InitializeBindings() public bindings!: Bindings;
+  @BindStateToController('tabManager')
+  @State()
+  private tabManagerState!: TabManagerState;
+  @Element()
+  private host!: HTMLElement;
+  public tabManager!: TabManager;
+
+  /**
+   * Whether to clear the filters when the active tab changes.
+   */
+  @Prop() clearFiltersOnTabChange?: boolean = false;
+
+  @State() public error!: Error;
+  @State() tabs: HTMLAtomicTabElement[] = [];
+
+  componentDidLoad() {
+    this.setInitialTab();
+  }
+
+  componentWillUpdate() {
+    const tabs = [...this.host.querySelectorAll('atomic-tab')];
+    this.tabs = tabs;
+
+    if (tabs.length === 0) {
+      this.error = new Error(
+        'The "atomic-tab-manager" element requires at least one "atomic-tab" child.'
+      );
+      return;
+    }
+  }
+
+  async setInitialTab() {
+    const initialTab = this.tabs[0];
+    const activeTab = getActiveTab(
+      this.bindings.engine.state.tabSet || {}
+    )?.tab;
+
+    if (initialTab && !activeTab) {
+      await initialTab.select(false);
```

Comment:
I think this pattern of having a boolean to control if it's "the first selection do not trigger a query" is because of a code smell.

Instead of having this pattern:

```html
atomic-tab-manager
   atomic-tab expression="foo"
   atomic-tab expression="bar"
atomic-tab-manager
```

Where: Each `atomic-tab` element creates its own tab controller, then "communicate" back to the manager, where the manager then has to copy the list and call a public method on the child `atomic-tab`, here what I would propose instead:

* atomic-tab-manager parse it's list of child element to read the list of expression/list of headless tab controller that should be created.

* atomic-tab-manager is in charge of creating N amount of headless tab controller depending on the above.

* Since it knows the full list of child element, the tab controller that should be active first (the first one in the DOM) gets a special initialization: `buildTabController(engine, {initialState: {isActive: true}`

* The render function of the manager become (pseudo code ):

```typescript
render() {
  <select>
     {this.allTabsController.map(tabController=> {
           <option value={tabController.id} onChange={()=> tabController.select()}>   
   })
   </select>
}
```